pkg/controller/garbagecollector/patch.go

```diff
@@ -30,12 +29,21 @@ import (
 )
 
 func deleteOwnerRefStrategicMergePatch(dependentUID types.UID, ownerUIDs ...types.UID) []byte {
```

Comment:
consider using a typed struct with json annotations to avoid map allocation. see the objectForFinalizersPatch type and use in this package as an example

Reply:
done